apps/users/models.py

```diff
@@ -361,7 +347,12 @@ def source(self):
 
     @property
     def name(self):
-        return smart_unicode(self.display_name or self.username)
+        if self.display_name:
+            return smart_unicode(self.display_name)
+        elif self.username.startswith('anonymous-'):
```

Comment:
This feels awkward because someone could legitimately have a username that starts with `anonymous-`. I guess it's not a big deal though.